Branch-SDK/Branch-SDK/BranchViewHandler.m

```diff
@@ -0,0 +1,125 @@
+//
+//  BranchViewHandler.m
+//  Branch-TestBed
+//
+//  Created by Sojan P.R. on 3/3/16.
+//  Copyright © 2016 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import "BranchViewhandler.h"
+#import "Branch.h"
+#import "BranchView.h"
+
+@interface BranchViewHandler() <UIWebViewDelegate>
+@end
+
+NSString * const BRANCH_VIEW_REDIRECT_SCHEME = @"branch-cta";
+NSString * const BRANCH_VIEW_REDIRECT_ACTION_ACCEPT = @"accept";
+NSString * const BRANCH_VIEW_REDIRECT_ACTION_CANCEL = @"cancel";
+
+@implementation BranchViewHandler
+
+static BranchViewHandler *branchViewHandler;
+BOOL isBranchViewAccepted = NO;
+NSString *currentActionName;
+
++ (BranchViewHandler *)getInstance {
+    if (!branchViewHandler) {
+        branchViewHandler = [[BranchViewHandler alloc] init];
+        branchViewHandler.branchViewCache = [[NSMutableArray alloc] init];
+    }
+    return branchViewHandler;
+}
+
+- (void)saveBranchViews:(NSArray *)branchViewList {
+    for (NSDictionary *branchViewItem in branchViewList) {
+        BranchView *branchView = [[BranchView alloc] initWithBranchView:branchViewItem];
+        [self.branchViewCache addObject:branchView];
+    }
+}
+
+- (BOOL)showBranchView:(NSString *)actionName withDelegate:(id)callback {
+    for (BranchView * branchView in self.branchViewCache) {
+        if ([branchView.branchViewAction isEqualToString:actionName] && [branchView isAvailable]) {
+            self.branchViewCallback = callback;
+            [self showView:branchView];
+            [branchView updateUsageCount];
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (void)showView:(BranchView *)branchView {
+    CGRect screenRect = [[UIScreen mainScreen] bounds];
+    UIWebView *webview = [[UIWebView alloc] initWithFrame:CGRectMake(0, 0, screenRect.size.width, screenRect.size.height)];
+    
+    webview.scrollView.scrollEnabled = NO;
+    webview.scrollView.bounces = NO;
+    webview.delegate = self;
+    
+    if (branchView.webHtml) {
+        [webview loadHTMLString:branchView.webHtml baseURL:nil];
+    }
+    else if (branchView.webUrl) {
+        NSURL *url = [NSURL URLWithString:branchView.webUrl];
+        NSURLRequest *requestObj = [NSURLRequest requestWithURL:url];
+        [webview loadRequest:requestObj];
+    }
+    else {
+        return;
+    }
+    
+    isBranchViewAccepted = NO;
+    currentActionName = branchView.branchViewAction;
+    
+    UIViewController *holderView = [[UIViewController alloc] init];
+    [holderView.view insertSubview:webview atIndex:0];
+    UIViewController *presentingViewController = [[[UIApplication sharedApplication].delegate window] rootViewController];
```

Comment:
one partner previously reported that using the `window` property crashed because their app delegate did not have a window property. Instead they've opened a PR asking for us to use `[[[UIApplication sharedApplication] windows] firstObject]`. Could you replace references to `window` with the above and ensure it still works?